### PR TITLE
New/connector puppeteer

### PR DESCRIPTION
**Pull request by molant**
## Pull request checklist

Make sure you:

- [x] Signed the [Contributor License Agreement](https://cla.js.foundation/webhintio/hint)
- [x] Followed the [commit message guidelines](https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages)

For non-trivial changes, please make sure you also:

- [x] Added/Updated related documentation.
- [x] Added/Updated related tests.

## Short description of the change(s)

Same as #2279 but with a cleaner history. I'll work on renaming to puppeteer and update the history so draft for now.

### CODE REVIEW

**Comment by molant** [comment]
I've rearranged the commit history and replace `chromium` by `puppeteer` everywhere 🤞
Hopefully Linux and mac builds will pass

**Comment by molant** [comment]
@antross I've done all the changes we talked about. Let me know if you want to do a joined PR review later and let's hope CI passes!

**Comment by molant** [comment]
@antross I've pushed the changes we discussed during the review. I'm missing the tests for the configuration that I'll try to get over later today after all the meetings.

File: packages/connector-puppeteer/src/connector.ts [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+
+        await this._engine.emitAsync('scan::end', { resource: this._finalHref });
+    }
+
+    public fetchContent(target: URL | string, customHeaders?: object): Promise<NetworkData> {
+        /*
+         * TODO: This should create a new tab, navigate to the
+         * resource and control what is received somehow via an event.
+         */
+        const assigns = compact([this && this._headers, customHeaders]);
```
Comment (antross) [inline packages/connector-puppeteer/src/connector.ts]: After digging into this a bit, I'm wondering if `this._headers` can/should be dropped entirely, both here and in the class as a whole. These are headers associated with the `target` request, but we're potentially applying them to requests for other resources as well which seems wrong. Plus most re-requests already pass the headers from the original request as part of `customHeaders`.

**Pull request closed without merging**